appsec/src/extension/tags.c

```diff
@@ -918,82 +956,219 @@ static PHP_FUNCTION(datadog_appsec_track_user_signup_event)
         return;
     }
 
+    zend_string *user_login = NULL;
     zend_string *user_id = NULL;
     HashTable *metadata = NULL;
-    zend_bool automated = false; // Don't document. Only internal usage
-    zend_bool copy_user_id = true;
-    if (zend_parse_parameters(ZEND_NUM_ARGS(), "S|hb", &user_id, &metadata,
-            &automated) == FAILURE) {
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "SS|h", &user_login, &user_id,
+            &metadata) == FAILURE) {
         mlog(dd_log_warning, "Unexpected parameter combination, expected "
-                             "(user_id, metadata)");
+                             "(user_login, user_id, metadata)");
         return;
     }
 
-    if (automated) {
-        user_collection_mode mode = dd_get_user_collection_mode();
-        if (mode == user_mode_disabled ||
-            !get_DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED()) {
-            return;
-        }
+    if (user_login == NULL || ZSTR_LEN(user_login) == 0) {
+        mlog(dd_log_warning, "Unexpected empty user login");
+        return;
+    }
 
-        if (mode == user_mode_anon) {
-            // Anonymize the user ID and ensure it isn't copied twice
-            user_id = dd_user_id_anonymize(user_id);
-            if (user_id == NULL) {
-                mlog(dd_log_debug, "Failed to anonymize user ID");
-                return;
-            }
+    zval *nullable meta = _root_span_get_meta();
+    if (!meta) {
+        return;
+    }
+
+    _user_event_triggered = true;
+    zend_array *meta_ht = Z_ARRVAL_P(meta);
+
+    user_collection_mode mode = dd_get_user_collection_mode();
+    if (mode == user_mode_disabled ||
+        !get_DD_APPSEC_AUTOMATED_USER_EVENTS_TRACKING_ENABLED()) {
+        return;
+    }
 
-            copy_user_id = false;
+    if (mode == user_mode_anon) {
+        user_id = dd_user_info_anonymize(user_id);
+        if (user_id == NULL) {
+            mlog(dd_log_debug, "Failed to anonymize user ID");
+            return;
         }
-    } else {
-        if (user_id == NULL || ZSTR_LEN(user_id) == 0) {
-            mlog(dd_log_warning, "Unexpected empty user id");
+
+        user_login = dd_user_info_anonymize(user_login);
+        if (user_login == NULL) {
+            mlog(dd_log_debug, "Failed to anonymize user login");
             return;
         }
     }
 
+    if (user_id && ZSTR_LEN(user_id) > 0) {
```

Comment:
According to the RFC I would say that user_id is optional, however I am not sure we should add the tags if the value provided is `null` or empty.

In any case I can make it optional in the function, it would indeed make more sense.